static/js/typeahead_helper.js

```diff
@@ -358,14 +359,31 @@ export function sort_slash_commands(matches, query) {
     return results.matches.concat(results.rest);
 }
 
-// Gives stream a score from 0 to 3 based on its activity
+// Gives stream a score from 0 to 5 based on its activity
+// stream that the message is addressed to (if any) > unmuted pinned streams > unmuted subscribed streams > muted streams > other streams
 function activity_score(sub) {
     let stream_score = 0;
     if (!sub.subscribed) {
         stream_score = -1;
     } else {
+        if (sub.name === compose_state.stream_name()) {
+            stream_score += 8;
+        }
         if (sub.pin_to_top) {
+            stream_score += 4;
+            if (!sub.is_muted) {
+                stream_score += 2;
+            } else {
+                stream_score += 1;
+            }
+        }
+        if (!sub.pin_to_top) {
             stream_score += 2;
+            if (!sub.is_muted) {
+                stream_score += 1;
+            } else {
+                stream_score += 0;
```

Comment:
Adding 0 seems pointless?

Reply:
Yes, I will remove that line and again try running the tests.

Reply:
@timabbott I have removed ``stream_score += 0`` and all tests are running successfully. Any further improvements?